esi_leap/tests/db/sqlalchemy/test_api.py

```diff
@@ -259,4 +266,122 @@ def test_offer_get_by_name(self):
 
     def test_offer_get_by_name_not_found(self):
 
-        assert api.offer_get_by_uuid('some_uuid') is None
+        self.assertEqual(api.offer_get_by_name('some_name'), [])
+
+    def test_offer_destroy(self):
+
+        offer = api.offer_create(test_offer_2)
+        api.offer_destroy(offer.uuid)
+        self.assertEqual(api.offer_get_by_uuid('offer_2'), None)
+
+    def test_offer_destroy_not_found(self):
+
+        self.assertEqual(api.offer_get_by_uuid('offer_4'), None)
+
+    def test_offer_update(self):
+
+        offer = api.offer_create(test_offer_3)
+        o = api.offer_create(test_offer_2)
+        values = {'start_time': o.start_time,
+                  'end_time': o.end_time}
+        api.offer_update(offer.uuid, values)
+        res = api.offer_get_by_uuid(offer.uuid)
+        self.assertEqual(res.start_time, offer.start_time)
+        self.assertEqual(res.end_time, offer.end_time)
+
+    def test_offer_get_all(self):
+
+        o = api.offer_create(test_offer_2)
+        offer = api.offer_create(test_offer_3)
+        res = api.offer_get_all({}).all()
+        self.assertEqual((o.to_dict(), offer.to_dict()),
+                         (res[0].to_dict(), res[1].to_dict()))
+
+    def test_contract_get_by_uuid(self):
+
+        o = api.offer_create(test_offer_2)
+        test_contract_4['offer_uuid'] = o.uuid
```

Comment:
Ah, don't do this; in the test_contract_4 dict, hardcode the offer_uuid key to the offer uuid (likely test_offer_4's uuid).

Reply:
ok ill fix that up!

Reply:
i remember having an issue with this working when we did it in the past. The problem was the uuid needed to belong to an offer that existed which was why we switched to doing it this way. Should I still cahnge it back?

Reply:
Ah, you're right - leave it for now!